### PR TITLE
Add code sharing sample

### PR DESCRIPTION
**Pull request by craiglabenz**
Begins introducing a sample for sharing code between the client and server. Functionally complete, but marked as `[WIP]` for now as it needs thorough documentation.

### CODE REVIEW

File: code_sharing/Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+run-server:
+	# Depends on `make build`
+	docker-compose up -d
+
+stop-server:
+	docker-compose stop
+
+# Flutter
+flutter-local:
+	cd client && flutter run -d macos
```
Comment (domesticmouse) [inline code_sharing/Makefile:11]: I feel like we should set this up for linux as well. 

And then there is Windows...

File: code_sharing/client/test/widget_test.dart [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (domesticmouse) [inline code_sharing/client/test/widget_test.dart:1]: Not even a smoke test to confirm it compiles? Sad face

**Review by domesticmouse — APPROVED** [review]

File: code_sharing/client/test/widget_test.dart (hunk 2 above)
Comment (craiglabenz) [reply code_sharing/client/test/widget_test.dart:1]: Oh right, that's fair!

File: code_sharing/Makefile (hunk 1 above)
Comment (craiglabenz) [reply code_sharing/Makefile:11]: The `Makefile` may actually just be a bad idea 🤔

File: tool/flutter_ci_script_stable.sh [hunk 3]
```diff
@@ -15,6 +15,7 @@ declare -ar PROJECT_NAMES=(
     "add_to_app/prebuilt_module/flutter_module"
     "android_splash_screen"
     "animations"
+    "code_sharing"
```
Comment (craiglabenz) [inline tool/flutter_ci_script_stable.sh:18]: @domesticmouse - After this commit, I was expecting to see my new sample audited in the CI pass, yet I'm not seeing it. Do you know if I forgot to make a change when adding this?

Comment (domesticmouse) [reply tool/flutter_ci_script_stable.sh:18]: This probably should be `"code_sharing/client"` and `"code_sharing/server"`

**Review by domesticmouse — APPROVED** [review]

File: tool/flutter_ci_script_stable.sh (hunk 3 above)
Comment (craiglabenz) [reply tool/flutter_ci_script_stable.sh:18]: omg, duh

File: tool/flutter_ci_script_beta.sh [hunk 4]
```diff
@@ -17,6 +17,7 @@ declare -ar PROJECT_NAMES=(
     "animations"
     "code_sharing/client"
     "code_sharing/server"
+    "code_sharing/server/shared"
```
Comment (domesticmouse) [inline tool/flutter_ci_script_beta.sh:20]: Do you need to add it to `master` as well?

**Review by domesticmouse — APPROVED** [review]

File: tool/flutter_ci_script_beta.sh (hunk 4 above)
Comment (domesticmouse) [reply tool/flutter_ci_script_beta.sh:20]: Probably want to pull all three into `stable` as well.

**Comment by domesticmouse** [comment]
Triple approved!

**Pull request merged**